### PR TITLE
10154 Type IAgent

### PR DESCRIPTION
**Pull request by twm**
## Scope and purpose

Align the types of the example strings in the IAgent docstring with the actual type (`bytes`). Also add some type annotations in the process.

Rendered: https://twisted--1569.org.readthedocs.build/en/1569/api/twisted.web.iweb.IAgent.html

## Contributor Checklist:

* [x] The associated ticket in Trac is here: https://twistedmatrix.com/trac/ticket/10154
* [x] I ran `tox -e lint` to format my patch to meet the [Twisted Coding Standard](https://twistedmatrix.com/documents/current/core/development/policy/coding-standard.html)
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: [News files](https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles))
* [x] The title of the PR starts with the associated Trac ticket number (without the `#` character).
* [x] I have updated the automated tests and checked that all checks for the PR are green.
* [x] I have submitted the associated Trac ticket for review by adding the word `review` to the keywords field in Trac, and putting a link to this PR in the comment; it shows up in https://twisted.reviews/ now.

### CODE REVIEW

**Review by adiroiban — APPROVED** [review]
Thanks for the update!

Only a minor comment and after the update this can be merged without another review.

Happy to see more typing usage.

I think that this PR can also remove the other docstring type definitions that are now duplicate:
 
* `@type headers: L{Headers} or L{None} `
* `@type bodyProducer: L{IBodyProducer} provider`
*  `@rtype: L{Deferred}`

https://github.com/twisted/twisted/pull/1569/files#diff-a6aecc8399511cbb1555514c8bc81e49cbd10bcfcae83b8629203e142045e25dL751

**Comment by twm** [comment]
Thanks again Adi! I made the changes you suggested. I will merge now as the Codecov failure seems to be some sort of malfunction.

**Pull request merged**
